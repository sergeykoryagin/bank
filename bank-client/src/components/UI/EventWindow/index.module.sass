@import "src/styles/mixins"

.button
    max-width: 400px
    width: 100%

.eventWindow
    display: flex
    flex-direction: column
    height: 100px
    max-width: 400px
    width: 100%
    padding: 16px
    background: $color-grey-300
    box-shadow: inset 4px 4px 24px #AFB8C7, inset -6px -6px 10px rgba(255, 255, 255, 0.45)
    overflow: scroll
    
.event
    margin-left: 20px
    @include font-p
    color: $color-blue-300
    