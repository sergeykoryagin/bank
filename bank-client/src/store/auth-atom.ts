import { atom } from 'recoil';

interface AuthState {
    isAuth: boolean;
    username: string | null;
    myId?: string | null;
}

const defaultAuthState: AuthState = {
    isAuth: false,
    username: null,
    myId: localStorage.getItem('myId'),
};

export const authAtom = atom<AuthState>({
    key: 'auth',
    default: defaultAuthState,
});
