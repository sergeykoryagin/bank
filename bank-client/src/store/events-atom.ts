import { atom } from 'recoil';

export const eventsAtom = atom<Array<string>>({
    key: 'events',
    default: [],
});
