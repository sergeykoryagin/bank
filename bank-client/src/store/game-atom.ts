import { GameResponse } from 'interfaces/game-response';
import { atom } from 'recoil';

export const gameAtom = atom<GameResponse | null>({
    key: 'game',
    default: null,
});
