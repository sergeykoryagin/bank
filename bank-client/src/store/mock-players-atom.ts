import { atom } from 'recoil';

export interface MockPlayer {
    id: string;
    username: string;
}

export const mockPlayersAtom = atom<Array<MockPlayer>>({
    key: 'test',
    default: [],
});
