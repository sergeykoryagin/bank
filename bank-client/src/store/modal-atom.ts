import { ReactElement } from 'react';
import { atom } from 'recoil';

export const modalAtom = atom<ReactElement | null>({
    key: 'modal',
    default: null,
});
