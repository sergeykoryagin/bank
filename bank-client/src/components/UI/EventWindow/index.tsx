import cn from 'classnames';
import { VFC } from 'react';
import styles from './index.module.sass';
import { useRecoilValue } from 'recoil';
import { eventsAtom } from 'store/events-atom';

interface Props {
    className?: string;
}

export const EventWindow: VFC<Props> = ({ className }: Props) => {
    const events = useRecoilValue(eventsAtom);
    return (
        <button className={cn(styles.button, className)}>
            <ul className={styles.eventWindow}>
                {events.map((event, index) => (
                    <li key={`${event}${index}`} className={styles.event}>
                        {event}
                    </li>
                ))}
            </ul>
        </button>
    );
};
