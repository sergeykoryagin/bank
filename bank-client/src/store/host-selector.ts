import { selector } from 'recoil';
import { authAtom } from 'store/auth-atom';
import { gameAtom } from 'store/game-atom';

export const hostSelector = selector({
    key: 'host',
    get: ({ get }) => {
        const game = get(gameAtom);
        const { myId } = get(authAtom);
        return game && game.hostId === myId;
    },
});
