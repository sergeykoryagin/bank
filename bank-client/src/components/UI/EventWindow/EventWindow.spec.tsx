import { fireEvent, render, screen } from '@testing-library/react';
import { EventWindow } from 'components/UI/EventWindow/index';
import { ReactElement, useEffect } from 'react';
import { RecoilRoot, useRecoilState } from 'recoil';
import { eventsAtom } from 'store/events-atom';

const renderWithRecoil = (component: ReactElement) => render(<RecoilRoot>{component}</RecoilRoot>);

const TestApp = () => {
    const [events, setEvents] = useRecoilState(eventsAtom);

    useEffect(() => {
        setEvents(['event №1', 'event №2', 'event №3']);
    }, []);

    const handleClick = () => {
        setEvents([...events, `event №${events.length}`]);
    };
    return (
        <div>
            <EventWindow />
            <button onClick={handleClick} />
        </div>
    );
};

describe('Input component', () => {
    it('should render correctly', () => {
        const { container } = renderWithRecoil(<EventWindow />);
        expect(container).toMatchSnapshot();
    });
    it('should render correctly', () => {
        renderWithRecoil(<TestApp />);
        expect(screen.getByText('event №1')).toBeInTheDocument();
        expect(screen.getByText('event №2')).toBeInTheDocument();
        expect(screen.getByText('event №3')).toBeInTheDocument();
    });
    // it('should render correctly', () => {
    //     const { container } = renderWithRecoil(<TestApp />);
    //     const ul = container.querySelector('ul');
    //
    //     const button = screen.getAllByRole('button');
    //
    //     fireEvent.click(button[0]);
    //
    //     expect(ul?.childElementCount).toBe(4);
    //     expect(screen.getByText('event №4')).toBeInTheDocument();
    // });
});
