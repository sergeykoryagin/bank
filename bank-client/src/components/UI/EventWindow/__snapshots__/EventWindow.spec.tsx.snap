// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render correctly 1`] = `
<div>
  <button
    class="button"
  >
    <ul
      class="eventWindow"
    />
  </button>
</div>
`;
